app/pages/routing/index/CountryOfRoutingPage.scala

```diff
@@ -35,12 +34,9 @@ case class CountryOfRoutingPage(index: Index) extends QuestionPage[Country] {
   override def toString: String = "countryOfRouting"
 
   override def route(userAnswers: UserAnswers, mode: Mode): Option[Call] =
-    Some(routes.CountryOfRoutingController.onPageLoad(userAnswers.lrn, mode, index))
-
-  override def cleanup(value: Option[Country], userAnswers: UserAnswers): Try[UserAnswers] =
-    value match {
-      case Some(_) => userAnswers.remove(TransitSection).flatMap(_.remove(ExitSection))
-      case None    => super.cleanup(value, userAnswers)
+    mode match {
+      case NormalMode => Some(routes.CountryOfRoutingController.onPageLoad(userAnswers.lrn, mode, index))
+      case CheckMode  => Some(controllers.routes.RouteDetailsAnswersController.onPageLoad(userAnswers.lrn))
```

Comment:
I think this can be reverted along with the `CountryOfRoutingNavigator` change, but would be good to understand why this change was made?